sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/collect.scala

```diff
@@ -65,8 +67,10 @@ abstract class Collect[T <: Growable[Any] with Iterable[Any]] extends TypedImper
     new GenericArrayData(buffer.toArray)
   }
 
+  lazy val typeChild = child.dataType
```

Comment:
Couple of things:
- Can you name this `bufferElementType` or something like that?
- Please make it protected.
- I would prefer to make this an abstract method (lazy val is not really needed) and make the subclass implement it.